app/models/task.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  id          :uuid             not null, primary key
+#  name        :string
+#  description :string
+#  state       :integer          default(0)
+#  project_id  :uuid
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_tasks_on_project_id  (project_id)
+#
+
+class Task < ActiveRecord::Base
+  belongs_to :project
+
+  enum state: [:todo, :in_progress, :done]
```

Comment:
This works well enough. I personally prefer to use the hash notation for enums though as it gives a little more flexibility both in values as well as ordering of the states (i.e. some folks like to alphabetize, which you can't do if you don't explicitly assign values and want to add a new one later not at the end).
Again, nothing wrong here, just pointing out the benefits of the alternative.